osquery/tables/system/darwin/system_profiler.mm

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/darwin/cfstring.h>
+#include <osquery/utils/darwin/system_profiler.h>
+#include <osquery/utils/json/json.h>
+
+#include <Foundation/Foundation.h>
+
+namespace osquery {
+namespace tables {
+
+id convertForJSON(id obj) {
+  @try {
+    if ([obj isKindOfClass:[NSDictionary class]]) {
+      NSMutableDictionary* dict = [NSMutableDictionary
+          dictionaryWithCapacity:[(NSDictionary*)obj count]];
+      for (id key in (NSDictionary*)obj) {
+        dict[key] = convertForJSON([(NSDictionary*)obj objectForKey:key]);
+      }
+      return dict;
+    } else if ([obj isKindOfClass:[NSArray class]]) {
+      NSMutableArray* arr =
+          [NSMutableArray arrayWithCapacity:[(NSArray*)obj count]];
+      for (id item in (NSArray*)obj) {
+        [arr addObject:convertForJSON(item)];
+      }
+      return arr;
+    } else if ([obj isKindOfClass:[NSDate class]]) {
+      // formatter will be autoreleased when the pool is drained
+      NSDateFormatter* formatter = [[NSDateFormatter alloc] init];
+      [formatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ssZ"];
+      NSString* result = [formatter stringFromDate:(NSDate*)obj];
+      return result;
+    } else if ([obj isKindOfClass:[NSData class]]) {
+      return [(NSData*)obj base64EncodedStringWithOptions:0];
+    } else if ([obj isKindOfClass:[NSURL class]]) {
+      return [(NSURL*)obj absoluteString];
+    } else if ([obj isKindOfClass:[NSUUID class]]) {
+      return [(NSUUID*)obj UUIDString];
+    } else if ([obj isKindOfClass:[NSNumber class]] ||
+               [obj isKindOfClass:[NSString class]]) {
+      // These types are already JSON-safe
+      return obj;
+    } else {
+      // For any other type, try to get a string representation
+      return [obj description];
+    }
+  } @catch (NSException* exception) {
+    LOG(WARNING) << "Exception in convertForJSON: "
+                 << [[exception reason] UTF8String];
+    return @"[Error converting object]";
```

Comment:
I think this is returning the error as a string? Does this make it hard to tell what's an error vs a string that happens to match?